proxy/proxy.go

```diff
@@ -891,6 +988,10 @@ func (p *Proxy) makeBackendRequest(ctx *context, requestContext stdlibcontext.Co
 
 	ctx.proxySpan.LogKV("http_roundtrip", EndEvent)
 	if err != nil {
+		if endpointMetrics != nil {
+			endpointMetrics.IncTotalFailedRoundTrips()
```

Comment:
What if there will be more than `2*statsResetPeriod` (2 minutes at the moment) between `endpointMetrics.IncTotalRequests()` and `endpointMetrics.IncTotalFailedRoundTrips()`.

In the current version of code, the total increment and failed round trips counter increment should increment the same slot corresponding to the same period of time. However, the slot for total counter would be rotated at the moment this line of code would be reached, wouldn't it? If so, the failed round trips counter will increase the slot for other time period (despite the fact this slot will be reached by the same memory address).

Reply:
IncTotalRequests could be done **after** round trip.